c_models/makefiles/extra.mk

```diff
@@ -0,0 +1,148 @@
+#
+LOG_DICT_FILES += $(LOG_DICT_FILE)
+
+# ----------------------------------------------------------------------
+# We need to tell the compliler to use the modified code.
+#
+CFLAGS += -I$(EXTRA_MODIFIED_DIR)
+
+# ----------------------------------------------------------------------
+# Make sure each neuron model has a unique build directory.
+#
+BUILD_DIR := $(dir $(EXTRA_MAKEFILE_PATH)))/../build/$(APP)
+
+# ----------------------------------------------------------------------
+# Support function for converting original source paths to modified paths.
+#
+# $(NEURAL_MODELLING_DIRS)/src is done by neural_build.mk so not needed here.
+# This function, if defined, is called from within convert_to_modified.
+#
+define extra_convert_to_modified
+$(patsubst $(EXTRA_SRC_DIR)%,$(EXTRA_MODIFIED_DIR)%,$(1))
+endef
+
+# ----------------------------------------------------------------------
+# Support function for converting modified C paths to object paths.
+#
+# $(NEURAL_MODELLING_DIRS)..... is done by neural_build.mk so not needed here.
+# This function, if defined, is called from within convert_to_object.
+#
+define extra_convert_to_object
+$(patsubst $(EXTRA_MODIFIED_DIR)%.c, $(BUILD_DIR)%.o, $(1))
+endef
+
+# ----------------------------------------------------------------------
+
+include $(NEURAL_MODELLING_DIRS)/makefiles/neuron/neural_build.mk
+
+# ----------------------------------------------------------------------
+
+# EXTRA_SYNAPSE_TYPE_OBJECTS is deprecated!
+# Instead add a new rule
+#$(BUILD_DIR).....o: $(EXTRA_MODIFIED_DIR)...c $(C_FILES_MODIFIED)
+#	-mkdir -p $(dir $@)
+#	$(SYNAPSE_TYPE_COMPILE) -o $@ make$<
+
+                       
+# EXTRA_STDP is deprecated
+# Instead add a new rule
+#$(BUILD_DIR).....o: $(EXTRA_MODIFIED_DIR)...c $(C_FILES_MODIFIED)
+#	-mkdir -p $(dir $@)
+#	$(STDP_COMPILE) -o $@ $<
+
+# ----------------------------------------------------------------------
+# Make rules to copy and transform the extra code (so that logging messages are
+# compressed on SpiNNaker).
+#
+$(EXTRA_MODIFIED_DIR)%.c: $(EXTRA_SRC_DIR)%.c
+	python -m spinn_utilities.make_tools.converter $(EXTRA_SRC_DIR) $(EXTRA_MODIFIED_DIR) $(EXTRA_LOG_DICT_FILE)
+
+$(EXTRA_MODIFIED_DIR)%.h: $(EXTRA_SRC_DIR)%.h
+	python -m spinn_utilities.make_tools.converter $(EXTRA_SRC_DIR) $(EXTRA_MODIFIED_DIR) $(EXTRA_LOG_DICT_FILE)
+
+$(EXTRA_LOG_DICT_FILE): $(EXTRA_SRC_DIR)
+	python -m spinn_utilities.make_tools.converter $(EXTRA_SRC_DIR) $(EXTRA_MODIFIED_DIR) $(EXTRA_LOG_DICT_FILE)
+
+# ----------------------------------------------------------------------
+# Build rule for the transformed extra source files.
+#
+$(BUILD_DIR)%.o: $(EXTRA_MODIFIED_DIR)%.c $(C_FILES_MODIFIED)
+	# extra build
+	-mkdir -p $(dir $@)
+	$(CC) $(CFLAGS) -o $@ $<
+
+# ----------------------------------------------------------------------
+# Add the directories containing the extra transformed sources to the clean
+# rule.
+#
+EXTRA_CLEAN_DIRS += $(EXTRA_MODIFIED_DIR)
```

Comment:
yet again, as these are buuilt by code i dont see, i shouldnt modify these

Reply:
You should not but at least now the makefile is understandable so you can/